packages/host/app/components/operator-mode/stack-item.gts

```diff
@@ -350,22 +351,26 @@ export default class OperatorModeStackItem extends Component<Signature> {
   }
 
   private doneEditing = restartableTask(async () => {
+    let item = this.args.item;
+    let { request } = item;
     // if the card is actually different do the save and dismiss, otherwise
     // just change the stack item's format to isolated
     if (this.hasUnsavedChanges) {
       // we dont want to have the user wait for the save to complete before
       // dismissing edit mode so intentionally not awaiting here
-      this.args.publicAPI.saveCard(this.card, true);
-    } else {
-      let { request } = this.args.item;
-      this.operatorModeStateService.replaceItemInStack(
-        this.args.item,
-        this.args.item.clone({
-          request,
-          format: 'isolated',
-        }),
-      );
+      let updatedCard = await this.args.saveCard(item.card);
```

Comment:
based on the comments above, do we really want to make the user wait for the save to complete before dismissing edit mode? is there a reason why we should `await`?

Reply:
Good point